src/components/user/UserView.jsx

```diff
@@ -0,0 +1,96 @@
+import { useRequest } from "ahooks";
+import { toast } from "react-hot-toast";
+
+import { USER_STATUS } from "@/constants";
+import { useAdmin } from "@/models";
+
+export default function UserView({ user, run }) {
+  const { approveUser, deleteUser } = useAdmin();
+
+  const approveUserRequest = useRequest(async () => approveUser(user.id), {
+    manual: true,
+    onSuccess: () => {
+      toast.success("Employer has been approved!");
+      run();
+    },
+    onError: () => {
+      toast.error("Failed to approve Employer");
+    },
+  });
+
+  const rejectUserRequest = useRequest(async () => deleteUser(user.id), {
+    manual: true,
+    onSuccess: () => {
+      toast.success("User has been removed!");
+      run();
```

Comment:
await !!!!!!!!
and we could display the toast after the reload

Reply:
Ah my bad !

And for the toast I think it does?